src/Common/Sitemap/SitemapIndex.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/**
+ * This file is part of the Zest Framework.
+ *
+ * @author   Muhammad Umer Farooq <lablnet01@gmail.com>
+ * @author-profile https://www.facebook.com/Muhammadumerfarooq01/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ * @since 3.0.0
+ *
+ * @license MIT
+ */
+
+namespace Zest\Common\Sitemap;
+
+class SitemapIndex extends SitemapWriter
+{
+    /**
+     * Start of sitemap.
+     *
+     * @since 3.0.0
+     *
+     * @var string
+     */
+    const START    = '<?xml version="1.0" encoding="UTF-8"?><sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">';
+
+    /**
+     * End of sitemap.
+     *
+     * @since 3.0.0
+     *
+     * @var string
+     */    
+    const END     = '</sitemapindex>';
+
+    /**
+     * Last modify.
+     *
+     * @since 3.0.0
+     *
+     * @var Datetime
+     */
+    private $lastMod;
+
+    /**
+     * Sitemap file.
+     *
+     * @since 3.0.0
+     *
+     * @var string
+     */
+    private $file;
+
+    /**
+     * XML structure.
+     *
+     * @since 3.0.0
+     *
+     * @var string
+     */    
+    private $raw = '
+    <sitemap>
+        <loc>:url</loc>
+        <lastmod>:lastmod</lastmod>
+    </sitemap>';
+
+    /**
+     * __construct.
+     * Specify file name wiht extension, the path will be the public
+     *
+     * @param (string) $file File name with extension (.xml).
+     *
+     * @since 3.0.0
+     *
+     * @return void
+    */    
+    public function __construct($file)
+    {
+        $this->file = route()->public.$file;
+        $this->lastMod = date('d:m:y');
+    }
+
+    /**
+     * Create the sitemap.
+     *
+     * @param (siring) $mode    Valid mode (for only access inside class)
+     * @param (string) $url     Valid url.
+     * @param (string) $lastMod Last modify.
+     *
+     * @since 3.0.0
+     *
+     * @return bool
+    */
+    private function create($mode, $url, $lastMod = null)
+    {
+        $lastMod    = $lastMod     ?: $this->lastMod;
+        $raw = str_replace([':url', ':lastmod'], [$url, $lastMod], $this->raw);
+        if ($mode === 'create') {
+            $fileH = new SitemapWriter($this->file, 'writeOnly');
+            $fileH->write(SitemapIndex::START.PHP_EOL);
+            $fileH->write($raw);
+            $fileH->write(PHP_EOL.SitemapIndex::END);
+        } elseif ($mode === 'append') {
```

Comment:
I don't know what coding style is about the `elseif` condition.

We use the `else if` or `elseif` to be the consistency.

Reply:
could you explain more please?

Reply:
I'm curious about the coding style.

If the `elseif` on all source code, it's fine to use this.

If not, please let these conditions be consistency.

Reply:
Oh i used `elseif` everywhere is this wrong???

I see this style also used by popular Frameworks like `Laravel`

Reply:
That's okay.

This issue will be resolved.